op-service/signer/xlayer_auth.go

```diff
@@ -0,0 +1,217 @@
+// If SecretKey or AccessKey is empty, no authentication is performed
+func (c *XLayerRemoteClient) addAuth(req *http.Request) error {
+	if c.config.SecretKey == "" || c.config.AccessKey == "" {
+		return nil
+	}
+
+	// Set accessKey header (use array syntax for consistency with xlayer-node)
+	req.Header[headerAccessKey] = []string{c.config.AccessKey}
+
+	// Generate signature using SHA256 algorithm
+	signature, err := c.genAuth(req, algoSha256)
+	if err != nil {
+		return fmt.Errorf("failed to generate signature: %w", err)
+	}
+
+	// Set sign header (use array syntax for consistency with xlayer-node)
+	req.Header[headerSignKey] = []string{signature}
+
+	return nil
+}
+
+// genAuth generates authentication signature from HTTP request
+func (c *XLayerRemoteClient) genAuth(req *http.Request, algorithm string) (string, error) {
+	// 1. Process URL query parameters
+	params := req.URL.Query()
+	treeMap := make(map[string][]string)
+	for _, v := range params {
+		var key string
+		for _, vv := range v {
+			key += vv
+		}
+		treeMap[key] = v
+	}
+
+	// 2. Read request body
+	var body strings.Builder
+	if req.Body != nil {
+		readCloser, err := req.GetBody()
+		if err != nil {
+			return "", fmt.Errorf("get body error: %w", err)
+		}
+		defer readCloser.Close()
+
+		buffer := make([]byte, 0)
+		// Read the request body into the buffer
+		_, err = io.Copy(&bufferWriter{&buffer}, readCloser)
+		if err != nil {
+			return "", fmt.Errorf("read body error: %w", err)
+		}
+
+		// Append the body if present
+		if len(buffer) != 0 {
+			body.Write(buffer)
+		}
+	}
+
+	// 3. Generate signature from treeMap and body
+	return c.generateSignature(treeMap, body.String(), algorithm)
+}
+
+// generateSignature generates signature from parameters and body
+func (c *XLayerRemoteClient) generateSignature(treeMap map[string][]string, body, algorithm string) (string, error) {
+	// Sort the map by values
+	var keys []string
+	for key := range treeMap {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	// Construct the content string
+	var content strings.Builder
+	for _, key := range keys {
+		values := treeMap[key]
+		for _, v := range values {
+			content.WriteString(v)
+		}
+	}
+
+	// Append the body if present
+	if body != "" {
+		content.WriteString(body)
+	}
+
+	// Calculate the hash based on the selected algorithm
+	var hash []byte
+	switch strings.ToLower(algorithm) {
+	case algoSha256:
+		hashObj := sha256.New()
+		hashObj.Write([]byte(content.String()))
+		hash = hashObj.Sum(nil)
+	default:
+		return "", fmt.Errorf("unsupported algorithm: %s", algorithm)
+	}
+
+	// Convert the hash to a hexadecimal string
+	hashString := hex.EncodeToString(hash)
+
+	// Encrypt the hash using AES
+	return encryptAES(hashString, c.config.SecretKey)
+}
+
+// encryptAES encrypts using AES ECB mode (consistent with xlayer-node)
+func encryptAES(src, key string) (string, error) {
+	block, err := aes.NewCipher([]byte(key))
```

Comment:
The key length is not validated, which may violate the AES key length requirements.